components/org.wso2.carbon.identity.data.publisher.authentication.audit/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/audit/AuthenticationAuditLoggingHandler.java

```diff
@@ -227,4 +237,11 @@ private boolean isAuthenticationAuditLoggingEnabled(Event event) throws Identity
 
         return false;
     }
+
+    private void addContextualInfo(String data) {
+
+        data += "\",\"" + USER_AGENT_KEY + "\" : \"" + MDC.get(USER_AGENT_QUERY_KEY)
```

Comment:
Can we use `StringBuilder` since we're concatenating a set of strings?

Reply:
One line concatenations are optimized and converted to `StringBuilder` under the hood. Memory wise is the same thing and to stick with the same pattern in the class I am avoiding the use of `StringBuilder` .